src/providers/authentication/authentication.ts

```diff
@@ -13,22 +14,43 @@ export class AuthenticationProvider {
   private employeeIdKey: string;
   private employeeId: string;
   private isUserAuthenticated: boolean;
+  private inUnAuthenticatedMode: boolean;
   public jwtDecode: any;
 
   constructor(
     private msAdal: MSAdal,
     private inAppBrowser: InAppBrowser,
-    appConfig: AppConfigProvider) {
-    this.authenticationSettings = appConfig.getAppConfig().authentication;
-    this.employeeIdKey = appConfig.getAppConfig().authentication.employeeIdKey;
+    private networkState: NetworkStateProvider,
+    private appConfig: AppConfigProvider) {
+  }
+
+  public initialiseAuthentication = ():void => {
+    this.authenticationSettings = this.appConfig.getAppConfig().authentication;
+    this.employeeIdKey = this.appConfig.getAppConfig().authentication.employeeIdKey;
     this.jwtDecode = jwtDecode;
     this.isUserAuthenticated = false;
+    this.inUnAuthenticatedMode = false;
+  }
+  public isInUnAuthenticatedMode = (): boolean => {
+    return this.inUnAuthenticatedMode;
   }
-
   public isAuthenticated = (): boolean => {
     return this.isUserAuthenticated;
   }
 
+  public setUnAuthenticatedMode = (mode: boolean): void => {
+    this.inUnAuthenticatedMode = mode;
+  }
+
+  public getUnAuthenticatedMode = (): boolean => this.inUnAuthenticatedMode;
```

Comment:
I guess this is the duplicated function. Also could we pick one style for these functions

Reply:
removed duplicate - and it is consistent with others.